File: src/java/EntityFX/src/EntityFX/Core/Whetstone/WhetstoneBenchmark.java
package EntityFX.Core.Whetstone;

import java.io.FileNotFoundException;
import java.io.IOException;

import EntityFX.Core.Writer;
import EntityFX.Core.Generic.BenchResult;
import EntityFX.Core.Generic.BenchmarkBase;

public class WhetstoneBenchmark extends BenchmarkBase<WhetstoneResult> {

    private Whetstone whetstone;

    public WhetstoneBenchmark(final Writer writer, final boolean printToConsole) throws FileNotFoundException {
        super(writer, printToConsole);
        whetstone = new Whetstone(printToConsole);
        Ratio = 1;
    }

    @Override
    public WhetstoneResult benchImplementation() throws IOException {
        return whetstone.bench(true);
    }

    /* error value type 'BenchResult' of method PopulateResult */
    @Override
    public BenchResult populateResult(final BenchResult benchResult, final WhetstoneResult dhrystoneResult) {
        benchResult.Result = dhrystoneResult.getMWIPS();
        benchResult.Points = dhrystoneResult.getMWIPS() * this.Ratio;
        benchResult.Units = "MWIPS";
        benchResult.Output = dhrystoneResult.getOutput();
        return benchResult;
    }

    @Override
    public void warmup(final Double aspect) {
    }
}

File: src/python/entityfx/whetstone_benchmark.py

from entityfx.benchmark_base import BenchmarkBase
from entityfx.whetstone import Whetstone
from entityfx.writer import Writer


class WhetstoneBenchmark(BenchmarkBase):
    
    def __init__(self, writer: Writer, print_to_console : bool=True, is_enabled : bool=True) -> None:
        super().__init__(writer, print_to_console, is_enabled)
        self.__whetstone = Whetstone(True)
        self.ratio = 1
    
    def benchImplementation(self):
        return self.__whetstone.bench()
    
    def populateResult(self, bench_result, dhrystone_result):
        bench_result["Points"] = dhrystone_result["MWIPS"] * self.ratio
        bench_result["Result"] = dhrystone_result["MWIPS"]
        bench_result["Units"] = "MWIPS"
        bench_result["Output"] = dhrystone_result["Output"]
        return bench_result
    
    def warmup(self, aspect : float=.05) -> None:
        pass
